src/coreclr/vm/codeman.cpp

```diff
@@ -1839,18 +1839,36 @@ CodeFragmentHeap::CodeFragmentHeap(LoaderAllocator * pAllocator, StubCodeBlockKi
 void CodeFragmentHeap::AddBlock(VOID * pMem, size_t dwSize)
 {
     LIMITED_METHOD_CONTRACT;
-    FreeBlock * pBlock = (FreeBlock *)pMem;
-    pBlock->m_pNext = m_pFreeBlocks;
-    pBlock->m_dwSize = dwSize;
-    m_pFreeBlocks = pBlock;
+
+    FreeBlock * pBlock = (FreeBlock *)malloc(sizeof(FreeBlock));
+    // In the OOM case we don't add the block to the list of free blocks
+    // as we are in a FORBID_FAULT code path.
```

Comment:
We typically use `new (nothrow)` instead of calling malloc directly.

Reply:
I cannot use new (nothrow) as it is called in the FORBID_FAULT code path and all the new operators with nothrow asserts on it:
https://github.com/dotnet/runtime/blob/c5caf17083bef58c9575677a1c02a78db06d5e36/src/coreclr/utilcode/clrhost_nodependencies.cpp#L350-L361

Reply:
That can be solved by `CONTRACT_VIOLATION` / `PERMANENT_CONTRACT_VIOLATION`

Reply:
Ah, I didn't know that, I'll do it that way then.